config/types/root_types.ts

```diff
@@ -10,63 +14,78 @@ export type ValueOf<ObjectType> = ObjectType[keyof ObjectType];
 export type NullableString = string | null;
 
 export interface TopLevelDocument {
-  authors: Array<AuthorProfile>,
-  score: number,
-  createdDate: string,
-  discussionCount: number,
-  unifiedDocument: UnifiedDocument,
-  hubs: Array<Hub>,
-  createdBy: CreatedBy | null,
-  userVote?: "downvote" | "upvote" | "neutralvote" | null,
-  title?: string,
-  externalUrl?: string,
-  doi?: string,
-  datePublished?: string,
-  journal?: string,
-  formats?: Array<PaperFormat>,
-  note?: any,
-  markdown?: string,
-  isReady: boolean,
-  consensus?: CitationConsensus,
-  boostAmount: number,
-  id: ID,
+  authors: Array<AuthorProfile>;
+  score: number;
+  createdDate: string;
+  discussionCount: number;
+  unifiedDocument: UnifiedDocument;
+  hubs: Array<Hub>;
+  createdBy: CreatedBy | null;
+  userVote?: "downvote" | "upvote" | "neutralvote" | null;
+  title?: string;
+  externalUrl?: string;
+  doi?: string;
+  datePublished?: string;
+  journal?: string;
+  formats?: Array<PaperFormat>;
+  note?: any;
+  markdown?: string;
+  isReady: boolean;
+  consensus?: CitationConsensus;
+  boostAmount: number;
+  id: ID;
 }
 
 export type PaperFormat = {
-  type: "pdf" | "latex",
-  url: string,
-}
+  type: "pdf" | "latex";
+  url: string;
+};
 
 export type UrlDocument = {
-  id?: ID,
-  title?: string,
-  slug?: string,
-  paperTitle?: string,
-  isRemoved?: boolean,
-}
+  id?: ID;
+  title?: string;
+  slug?: string;
+  paperTitle?: string;
+  isRemoved?: boolean;
+};
+
+export type RhDocumentType =
+  | "eln"
+  | "hypothesis"
+  | "paper"
+  | "post"
+  | "researchhub_posts";
+export type GrmVoteType = "downvote" | "neutralvote" | "upvote";
```

Comment:
I'm not sure the GRM prefix is needed. I think it will be confusing to new developers joining the team. I think the GRM part is purely a backend construct. From the FE perspective should probably just be `VoteType`